internal/crawler/crawl.go

```diff
@@ -62,13 +66,16 @@ type SummaryJSON struct {
 // provided configuration. Returns the number of issues that were found.
 func Crawl(ctx context.Context, config Config, stdout, stderr io.Writer) (int64, error) {
 	identifier := crawl.Log4jIdentifier{
-		ZipWalker:          archive.WalkZipFiles,
-		TarWalker:          archive.WalkTarFiles,
-		Limiter:            limiterFromConfig(config.ArchivesCrawledPerSecond),
-		ArchiveWalkTimeout: config.ArchiveListTimeout,
-		OpenFileZipReader:  zip.OpenReader,
-		ArchiveMaxDepth:    config.ArchiveMaxDepth,
-		ArchiveMaxSize:     config.ArchiveMaxSize,
+		ZipWalker:                          archive.WalkZipFiles,
+		TarWalker:                          archive.WalkTarFiles,
+		Limiter:                            limiterFromConfig(config.ArchivesCrawledPerSecond),
+		ArchiveWalkTimeout:                 config.ArchiveListTimeout,
+		OpenFileZipReader:                  zip.OpenReader,
+		ArchiveMaxDepth:                    config.ArchiveMaxDepth,
+		ArchiveMaxSize:                     config.ArchiveMaxSize,
+		IdentifyObfuscation:                config.ObfuscatedClassNameAverageLength > 0 && config.ObfuscatedPackageNameAverageLength > 0,
+		ObfuscatedClassNameAverageLength:   float32(config.ObfuscatedClassNameAverageLength),
```

Comment:
We can leave as-is for this PR, but I think that a better approach for the crawler would be to have it take something like an `ObfuscationDetector` and to instantiate it based on the flags. I would say a similar thing for archive settings etc.

Basically, right now the top-level flags are pushed all the way down into the structs as well, whereas I think for usability/understandability/use in other code, it generally works better to have structured config and convert from flags to the structured config before calling in.

This is basically an FYI/possible TODO for myself later -- no need to change as part of this PR.